homeassistant/components/aprilaire/select.py

```diff
@@ -0,0 +1,199 @@
+"""The Aprilaire select component."""
+
+from __future__ import annotations
+
+from pyaprilaire.const import Attribute
+
+from homeassistant.components.select import SelectEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import AprilaireCoordinator
+from .entity import BaseAprilaireEntity
+
+AIR_CLEANING_EVENT_MAP = {0: "off", 3: "event_clean", 4: "allergies"}
+AIR_CLEANING_MODE_MAP = {0: "off", 1: "constant_clean", 2: "automatic"}
+FRESH_AIR_EVENT_MAP = {0: "off", 2: "3hour", 3: "24hour"}
+FRESH_AIR_MODE_MAP = {0: "off", 1: "automatic"}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Aprilaire select devices."""
+
+    coordinator: AprilaireCoordinator = hass.data[DOMAIN][config_entry.unique_id]
+
+    assert config_entry.unique_id is not None
+
+    entities: list[SelectEntity] = []
+
+    if coordinator.data.get(Attribute.AIR_CLEANING_AVAILABLE) == 1:
+        entities.append(
+            AprilaireAirCleaningEventEntity(coordinator, config_entry.unique_id)
+        )
+
+        entities.append(
+            AprilaireAirCleaningModeEntity(coordinator, config_entry.unique_id)
+        )
+
+    if coordinator.data.get(Attribute.VENTILATION_AVAILABLE) == 1:
+        entities.append(
+            AprilaireFreshAirEventEntity(coordinator, config_entry.unique_id)
+        )
+
+        entities.append(
+            AprilaireFreshAirModeEntity(coordinator, config_entry.unique_id)
+        )
+
+    async_add_entities(entities)
+
+
+class AprilaireAirCleaningEventEntity(BaseAprilaireEntity, SelectEntity):
+    """Aprilaire air cleaning event entity."""
+
+    _attr_translation_key = "air_cleaning_event"
+
+    def __init__(self, coordinator: AprilaireCoordinator, unique_id: str) -> None:
+        """Initialize an Aprilaire air cleaning event entity."""
+
+        super().__init__(coordinator, unique_id)
```

Comment:
What do we do with the unique id? Do we append something with it?

Reply:
It is used to generate the `_attr_unique_id` on the base entity. This was how it was decided to be done in the initial implementation of the integration.